### ISSUE TITLE
pipe in boolean property causes errors

### ISSUE THREAD

**Issue by BibbyChung** [kosich/rxjs-proxify#11]
hi there,

i can't use pipe method in  boolean property. i try to fix it, but for me it's too difficult. please fix this. thanks ^^ 

``` javascript
const state = proxify(new BehaviorSubject<{ a: number, b: string, c: boolean }>({ a: 0, b: '999', c: true }));
state.a.pipe(map(a => a));
state.b.pipe(map(a => a));

state.c.pipe(map(a => a));
```

![image](https://user-images.githubusercontent.com/8520661/121382687-6ead2980-c979-11eb-91fd-0f9ea10c7594.png)

**Comment by kosich** [comment]
Hey! Thanks for reporting this!

Looks quite weird. Alas, I don't have a solution atm. Will investigate and ping back

**Comment by kosich** [comment]
Heya!

Seems like TS split `boolean` intro `true | false`, so that we got `Proxy<true> | Proxy<false>`. This merged `.pipe` overloads and eliminated all but no args version `.pipe()` — therefore the TS complaint in your screenshot. It happened when we conditioned between callable and non-callable ObservableProxy, though I don't fully understand the underlying TS processes.

Fixed in `0.1.1`

```
npm i rxjs-proxify@latest
```

Cheers! 🙂
